tests/util/db_connection.py

```diff
@@ -1,19 +1,32 @@
 from __future__ import annotations
 
+import random
+import string
 import tempfile
 from contextlib import asynccontextmanager
 from pathlib import Path
-from typing import AsyncIterator
+from typing import AsyncIterator, Union
 
 from chia.util.db_wrapper import DBWrapper2
 
 
+def gen_in_memory_db_uri(prefix: str) -> str:
+    random_string = "".join(random.choice(string.ascii_letters) for _ in range(8))
+    return f"file:{prefix}{random_string}?mode=memory&cache=shared"
+
+
 @asynccontextmanager
-async def DBConnection(db_version: int) -> AsyncIterator[DBWrapper2]:
-    with tempfile.TemporaryDirectory() as directory:
-        db_path = Path(directory).joinpath("db.sqlite")
-        _db_wrapper = await DBWrapper2.create(database=db_path, reader_count=4, db_version=db_version)
-        try:
-            yield _db_wrapper
-        finally:
-            await _db_wrapper.close()
+async def DBConnection(db_version: int, in_memory_db: bool = True) -> AsyncIterator[DBWrapper2]:
+    if in_memory_db:
+        directory = None
+        db_path: Union[str, Path] = gen_in_memory_db_uri("db_cnx_")
```

Comment:
if multiple tests open a DB connection to the same URI, won't they access the same underlying DB, and possibly interfere with each other?